### PR TITLE
Separate out parsers

### PR DESCRIPTION
**Pull request by vagishagupta23**
Parsers are separated out of the `main.py` module to a new `parsers.py`
module to make code cleaner and compact.

This patch is heavily based on Vidushi's work.
Name: Vidushi Agrawal 
Email id: <vidushi229@gmail.com>

Make sure these boxes are signed before submitting your Pull Request
-- thank you.

- [x] I have read the contributing guide lines at
  https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation
  contribution agreement at
  https://suricata-ids.org/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the
  changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/2874

Describe changes:
-  Created a new module `parsers.py` with reduced repetitions of 
   `add_argument`  by storing the arguments in a tuple and added
    the parsers to the loop.
- Also, `parsers.py` is broken into smaller functions based on different
   parsers i.e. `global_parser`, `update_parsers` and `commands`.

### CODE REVIEW

File: suricata/update/parsers.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+        (("-T", "--test-command"),
+            {'metavar': '<command>',
+                'help': "Command to test Suricata configuration"}),
+        (("--no-test",),
+            {'action': 'store_true', 'default': False,
+                'help': "Disable testing rules with Suricata"}),
+        (("--no-merge",),
+            {'action': 'store_true', 'default': False,
+                'help': "Do not merge the rules into a single file"})
+    ]
```
Comment (shivan1b) [inline suricata/update/parsers.py]: I think we should take this data structure and others out of the functions. Functions can be compact that way. At present, the function is mostly about declaration.

Comment (vagishagupta23) [reply suricata/update/parsers.py]: @shivan1b I didn't very well understand it. If we move out the data structure, then what should we keep inside the function?

**Comment by vagishagupta23** [comment]
> @vagishagupta23 when we give credits to someone, we usually mention them like this: "Thanks to Vagisha Gupta [vagishagupta23@gmail.com](mailto:vagishagupta23@gmail.com)" as it shows up in git Author.
> 
Yes, this looks so much better. Thanks a lot!

File: suricata/update/parsers.py (hunk 1 above)
Comment (shivan1b) [reply suricata/update/parsers.py]: The logic. We can move this list out of the function and keep the operations on the list inside the function.

Comment (vagishagupta23) [reply suricata/update/parsers.py]: Yes, I get that. We can do this with `parse_global` and `parse_update` functions. 
Does `parse_commands` looks fine to you?

**Comment by vagishagupta23** [comment]
@shivan1b For this patch `"-V", "--version"` doesn't work. Fixing it with other requested changes.

**Comment by vagishagupta23** [comment]
Created a new PR #167.

**Pull request closed without merging**